Project.toml

```diff
@@ -15,6 +15,7 @@ Graphs = "86223c79-3864-5bf0-83f7-82e725a168b6"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 Requires = "ae029012-a4dd-5104-9daa-d747884805df"
 SparseArrays = "2f01184e-e22b-5df5-ae63-d93ebab69eaf"
+SparsityDetection = "684fba80-ace3-11e9-3d08-3bc7ed6f96df"
```

Comment:
Use Symbolics for this now.

Reply:
I tried, following the examples in Symbolics, but it's not working as I'd expect:
```julia
julia> using Symbolics, LinearAlgebra

julia> f(x) = -dot(x, x) + 2 * x[2] * x[3]^2;

julia> @variables u[1:5];

julia> ff = f(u)
2(u[3]^2)*u[2] - (u[1]^2) - (u[2]^2) - (u[3]^2) - (u[4]^2) - (u[5]^2)

julia> Symbolics.hessian_sparsity(ff, u)
ERROR: TypeError: non-boolean (Num) used in boolean context
[...]
```
Also tried this, which doesn't seem to work either:
```julia
julia> Symbolics.hessian(ff, u) # all zeros, wrong
5×5 Matrix{Num}:
 0  0  0  0  0
 0  0  0  0  0
 0  0  0  0  0
 0  0  0  0  0
 0  0  0  0  0

julia> ForwardDiff.hessian(f, ones(5)) # correct
5×5 Matrix{Float64}:
 -2.0   0.0  0.0   0.0   0.0
  0.0  -2.0  4.0   0.0   0.0
  0.0   4.0  2.0   0.0   0.0
  0.0   0.0  0.0  -2.0   0.0
  0.0   0.0  0.0   0.0  -2.0
```
Am I just using Symbolics incorrectly?

Reply:
Use `collect(u)` first?

Reply:
👍 Thanks, got it!

However, I now see SparsityDetection is already a dependency for some of the existing tests, which is also all I'm using it for (I added it to the main [deps] section of Project.toml by mistake). Unless you feel strongly about switching to Symbolics here, I think it makes more sense to just use the current dependency on SparsityDetection.